cmd/sales-api/handlers/user.go

```diff
@@ -54,6 +54,18 @@ func (u *User) Retrieve(ctx context.Context, log *log.Logger, w http.ResponseWri
 	dbConn := u.MasterDB.Copy()
 	defer dbConn.Close()
 
+	claims, ok := ctx.Value(auth.Key).(auth.Claims)
+	if !ok {
+		return web.ErrUnauthorized
+	}
+
+	// Regular users are only able to see their own record. Admins can see
+	// anyone's record. So therefore if the caller is asking for someone else and
+	// they don't have the admin role then they're forbidden.
+	if claims.Subject != params["id"] && !claims.HasRole(auth.RoleAdmin) {
```

Comment:
We need to talk about this. I am concerned things are not where they need to be. This might be right but my gut is saying this is not the correct place for things. We need to jump on a review next week.